app/routes.js

```diff
@@ -1,10 +1,11 @@
 const user = require('./controllers/user'),
+  auth = require('./middlewares/auth'),
   userValidator = require('./middlewares/user');
 
 exports.init = (app) => {
   
   app.post('/users', [], user.create);
-
   app.post('/users/sessions', userValidator.validateLoginInput, user.signin);
+  app.get('/users/list/:offset?/:limit?', auth.checkCredentials, user.list);
```

Comment:
We usually try to make our routes the more Restfully we can. In this case the list keyword would not be necessary. Also the offset and limit params we handle them as query params instead of adding them to the route itself.

In particular these case where we want to obtain a list of user the route should be:
GET /users
The final s implies that we want a collection of users and not only one, and after that any additional param can be added as query param by the one consuming the endpoint.
For example someone that wants the first page with 25 elements could send:
GET /users?offset=0&limit=25
or if someone wants the 3 page with 10 elements
GET /users?offset=20&limit=10
or if someone wants only to send offset (and use a default page limit)
GET /users?offset=100
or if these endpoints also has other params like, order or filtering by name you could send:
GET /users?order=age&name=Kevin&limit=10&offset=0

So this way you should only add the logic of handling that params in the controller and not in the route.
Did I explain myself? Otherwise let me know and we talk about this a little bit more in Slack or I can share you some posts about Restful APIs.

Reply:
Remember I told you the `list`part of the route is fine because the card asked for it, but the part of the offset and limit is still wrong.